### DISCUSSION TITLE
rAudio 20230525: Webradio -> "Change cover art" from default to custom not possible

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1320] [Bug Reports]
1. I created a new folder for webradios (GUI)
2. I inserted several webradios into the new folder (GUI)
3. After that I tried to use the "Change cover art" function to define custom cover arts, This was not possible, no reaction from GUI.

Investigating the problem I detected the following possible causes:
1. The new folder had ownership root:root, all other folders: http:http
2.  Within the new folder some (not all) webradios had ownership root:root instead of http:http

Applying appropriate chmods did not solve the problem.

**Comment by klassikmann** [comment]
rAudio 20230528: Problem still present.

**Comment by klassikmann** [comment]
Still present in 20230609.

**Reply by rern** [reply]
Please try:
`wget https://github.com/rern/rAudio/raw/UPDATE/srv/http/assets/js/context.js -O /srv/http/assets/js/context.js`

**Reply by klassikmann** [reply]
@rern 
No change. 
Guess: Creating a new webradio does not generate a default coverart in img-directory?

**Reply by klassikmann** [reply]
> @rern No change. Guess: Creating a new webradio does not generate a default coverart in img-directory?

Sorry. After a reboot the problem is fixed.

**Reply by klassikmann** [reply]
> > @rern No change. Guess: Creating a new webradio does not generate a default coverart in img-directory?
> 
> Sorry. After a reboot the problem is fixed.

Add on: But ownership is root:root instead of http:http.

**Comment by klassikmann** [comment]
**rAudio 20230615**:
Observations:
1. Change coverart from default icon: o.k.
2. If an existing webradio is deleted, the corresponding JPGs in webradio/img retained.
3. If a new webradio is created and a custom coverart is assigned, the ownership of the webradio entry and the corresponding images is root:root and not http:http as before.

**Comment by klassikmann** [comment]
The release notes for rAudio 20230616 contains:
`• Fix - Web Radio - Delete coverarts when stations deleted`
I cannot confirm the statement. After deleting a webradio, the coverarts are still present. 

And the 3. observation from my last posting
>     3. If a new webradio is created and a custom coverart is assigned, the ownership of the webradio 
>     entry and the corresponding images is root:root and not http:http as before.
is still valid.

**Comment by klassikmann** [comment]
Observations in **rAudio 20230620**:

**New webradio**
Created in `/srv/http/data/webradio` with ownership http:http

**Change coverart**
Two jpg-files created in `/srv/http/data/webradio/img` with ownership root:root, not http:http
@rern Intended behaviour (ownership)?

**Delete webradio**
Entry in `/srv/http/data/webradio` and two entries in `/srv/http/data/webradio/img` deleted

**Comment by klassikmann** [comment]
Observations in rAudio 202306230:

**Delete webradio**
Changed behaviour: Entry in `/srv/http/data/webradio` deleted but the two entries in `/srv/http/data/webradio/img` not deleted.
Consequence: After recreating the same webradio the coverarts are present immediately.
@rern Intended behaviour?

Test scenario "Manually delete the coverarts after delete of corresponding webradio":
1. Recreate same webradio
Created in `/srv/http/data/webradio` with ownership http:http
2. Change coverart 
Two jpg-files created in `/srv/http/data/webradio/img` with ownership root:root, not http:http

**Reply by rern** [reply]
Add
- `http://stream.live.vc.bbcmedia.co.uk/bbc_world_service`

Context menu
- Change cover art - OK
- Delete - Both the station and cover art files are gone


What was your Web Radio URL?

**Comment by klassikmann** [comment]
> Add
> 
>     * http://stream.live.vc.bbcmedia.co.uk/bbc_world_service
> 
> 
> Context menu
> 
>     * Change cover art - OK
> 
>     * Delete - Both the station and cover art files are gone
Same here, coverarts ownership are root:root instead of http:http in the past.

> What was your Web Radio URL?

SuperStereo 1 - Yacht Rock, New wave, Adult Pop
http://198.204.228.202:8160/flac

**Reply by rern** [reply]
Tried with the URL. 
Delete removed station and cover art files without the leftover issue.

**Comment by klassikmann** [comment]
> Tried with the URL. Delete removed station and cover art files without the leftover issue.

@rern 
New testing with other webradios confirmed the behavior I described (leftover of coverarts). I leave the issue open and continue testing locally. I will get back to you with any new insights.
